### ISSUE TITLE
Colour contrast for public avatars

### ISSUE THREAD

**Issue by nathan-at-jisc** [libscie/ResearchEquals.com#219]
Hi,

My name's Nathan Sainsbury and I work at Jisc. One of our new projects is Octopus (which you may have heard of).

What you've done so far looks really good and the codebase looks great.

Upon creating an account, I was auto generated a [profile image](https://eu.ui-avatars.com/api/?rounded=true&background=6d0aac&name=nathansainsbury).

I can see that this is auto generated, and just wanted to flag that the colour contrast doesn't pass WCAG AA or AAA (with a contrast ratio of: 1.73:1).

It might be that it's not a priority, but wanted to flag just in case.

Was going to raise a PR to see if I could fix this, and happy to do so if that would help.

----
Nathan Sainsbury
[@nathansains](https://twitter.com/nathansains)

**Comment by nathan-at-jisc** [comment]
A possible solution could be to also specifically reference `color` as well as `background` and make sure it has a good color contrast? i.e rather than

```js
create: {
  handle,
  avatar: `https://eu.ui-avatars.com/api/?rounded=true&color={color}&background=${hexColor}&name=${handle}`,
}
```

**Comment by chartgerink** [comment]
Hi @nathan-at-jisc 👋 Welcome to the ResearchEquals repo - thanks for opening up your first issue 😊 

This is a good point! I hoped https://eu.ui-avatars.com/ would be better at auto-determining the contrast. It got swamped in all the other things, and I agree this is more important than I've given it attention.

Did you have in mind using something like [`color-contrast-calc`](https://www.npmjs.com/package/color-contrast-calc) to ensure the contrast of the text and background?

**Comment by nathan-at-jisc** [comment]
Yeah, was thinking something like that. Completely understand that there's loads going on, can see the amount of work going into this, really impressive.

I'd be happy to raise a PR and give it a go, wasn't sure if that would come across passive aggressive though 😆 .

**Comment by chartgerink** [comment]
Was looking at contrast for module color selection and found this function on [StackOverflow](https://stackoverflow.com/questions/3942878/how-to-decide-font-color-in-white-or-black-depending-on-background-color):

```js
function pickTextColorBasedOnBgColorAdvanced(bgColor, lightColor, darkColor) {
  var color = (bgColor.charAt(0) === '#') ? bgColor.substring(1, 7) : bgColor;
  var r = parseInt(color.substring(0, 2), 16); // hexToR
  var g = parseInt(color.substring(2, 4), 16); // hexToG
  var b = parseInt(color.substring(4, 6), 16); // hexToB
  var uicolors = [r / 255, g / 255, b / 255];
  var c = uicolors.map((col) => {
    if (col <= 0.03928) {
      return col / 12.92;
    }
    return Math.pow((col + 0.055) / 1.055, 2.4);
  });
  var L = (0.2126 * c[0]) + (0.7152 * c[1]) + (0.0722 * c[2]);
  return (L > 0.179) ? darkColor : lightColor;
}
```

Thought it might be of help here too. 😊

**Comment by chartgerink** [comment]
@all-contributors add @nathan-at-jisc for bug
